addons/product/models/product.py

```diff
@@ -306,6 +251,26 @@ def _compute_product_template_attribute_value_ids(self):
                 else:
                     product.product_template_attribute_value_ids += values_per_template[product.product_tmpl_id.id][pav.id]
 
+    @api.multi
+    def _get_image_fallback_record(self):
+        """Override to get the image from the template if no image is set on the
+        variant."""
+        return self.product_tmpl_id
+
+    @api.multi
+    def _force_write_image_on_fallback(self):
+        """Override to always write on the template if there is only one
+        variant.
+
+        This is needed because when there is only one variant, the user doesn't
+        know there is a difference between template and variant, he expects both
+        images to be the same.
+        """
+        return self.env['product.product'].search_count([
```

Comment:
active_test=True

Reply:
Isn't that exactly the same as adding active=True in the domain? Isn't it better to not change the context if avoidable? To keep the same env for prefetch etc.